v6/technical/filtering-generic.md

```diff
@@ -0,0 +1,203 @@
+- The **availability** of this filter functionality is determined by each implementing
+  organisation. The general structure and semantics are
+  **standardised** within the OOAPI specification.  
+- This mechanism is intended to promote **consistency** across implementations, while
+  allowing flexibility in technical realisation.  
+- Filtering can complement **[data minimisation](technical/data-minimisation.md)** principles, as it allows clients to
+  request only relevant data subsets.  
+- Consumers or working groups that wish to apply **generic filtering mechanisms** are
+  encouraged to use this approach for the sake of consistency across implementations.  
+- The structure is **extensible**. New operations or fields can be introduced if they
+  follow the same serialisation pattern.  
+- Filters are a **convenience feature** and not part of the formal API contract.
+
+---
+
+When requesting collections of resources, a client can optionally request items to be
+filtered using the query parameter `filter_query`.
+
+This mechanism is a **generic structure** that enables flexible filtering on any exposed
+field of a resource. It is **not required** and **may not be implemented** by every
+implementing organisation. When unsupported, the parameter is simply ignored.
+
+---
+
+## Generic filter structure using `filter_query`
+
+The `filter_query` parameter allows clients to express conditions on fields in a
+structured way using *deepObject* serialisation.
+
+### Example pattern
+
+```http
+filter_query[field][operation]=value
+```
+
+Each condition defines one filter on a specific field. When multiple filters are present,
+they are combined using logical **AND**.
+
+### Example request
+
+```http
+GET /organisations/{organisationId}/course-offerings
+  ?filter_query[programme_offerings.programme.primary_code][in]=B-IT-2025
+  &filter_query[enrol_start_date_time][gt_date]=2025-08-01T00:00:00Z
+```
+
+This request returns all course offerings from an organisation with organisationId that starts 
+after 1 August 2025 and belong to the programme with code `B-IT-2025`.
+
+Additional fields can be filtered in the same way. For instance:
+
+```http
+GET /persons?filter_query[surname][like]=Sm*&filter_query[dateOfBirth][eq_date]=1998-05-12
+```
+
+This example returns all persons whose surname starts with “Sm” and whose date of
+birth is exactly 12 May 1998.
+
+---
+
+## Using wildcards
+
+Partial matches can be expressed using wildcards. Implementing organisations **SHOULD**
+use only the asterisk `*` as a wildcard. Other characters such as `%` **must be escaped**
+and **may not be supported** consistently.
+
+### Example request
+
+```http
+GET /courses?filter_query[name[].value][like]=bio*
+```
+
+This request returns all items whose `name` starts with “bio”.
+
+If the server also supports the `%` wildcard (for example, in SQL-style pattern matching),
+the following request may produce the same result:
+
+```http
```

Comment:
Do we really want to support %? and not just try to move all users to * for simplicity?